From ccb3b997d48475b645ee6939705c7e3137300194 Mon Sep 17 00:00:00 2001
From: Guo Mengqi <guomengqi3@huawei.com>
Date: Tue, 13 Sep 2022 10:48:24 +0800
Subject: [PATCH 0208/1076] [Huawei] mm: sharepool: check size=0 in
 mg_sp_make_share_k2u()

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5QQPG

--------------------------------

Add a size-0-check in mg_sp_make_share_k2u() to avoid passing 0-size spa
to __insert_sp_area().

Signed-off-by: Guo Mengqi <guomengqi3@huawei.com>
---
 mm/share_pool.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index 68a89ec15eeb..95c90ed8af30 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -2871,6 +2871,11 @@ static int sp_k2u_prepare(unsigned long kva, unsigned long size,
 	unsigned int page_size = PAGE_SIZE;
 	unsigned long kva_aligned, size_aligned;
 
+	if (!size) {
+		pr_err_ratelimited("k2u input size is 0.\n");
+		return -EINVAL;
+	}
+
 	if (sp_flags & ~SP_FLAG_MASK) {
 		pr_err_ratelimited("k2u sp_flags %lx error\n", sp_flags);
 		return -EINVAL;
-- 
2.17.1

